synapse/http/matrixfederationclient.py

```diff
@@ -66,6 +65,19 @@
     MAXINT = sys.maxint
 
 
+class ProxyMatrixFederationEndpointFactory(object):
+    def __init__(self, hs):
+        self.reactor = hs.get_reactor()
+        self.tls_client_options_factory = hs.tls_client_options_factory
+        self.gateway_addr = hs.config.federation_request_gateway_addr
+
+    def endpointForURI(self, uri):
+        return matrix_federation_endpoint(
```

Comment:
this is going to look up the SRV record for the proxy, I think. I don't think that's what you want to do.